hibernate-reactive-core/src/test/java/org/hibernate/reactive/types/JavaTypesArrayTest.java

```diff
@@ -62,12 +104,48 @@ private void testField(
 	}
 
 	@Test
-	public void testStringArrayType(VertxTestContext context) {
+	public void testStringArrayTypeWithNoAnnotations(VertxTestContext context) {
+		Basic basic = new Basic();
+		String[] dataArray = { "Hello world!", "Hello earth" };
+		basic.stringArrayNoAnnotations = dataArray;
+
+		testField( context, basic, found -> {
+			assertArrayEquals( dataArray, found.stringArrayNoAnnotations );
+			// PostgreSQL is the only DB that changes it's `create table...` statement to include array information
+			// This test checks that the logged query is correct and contains "array[100]"
+			if ( dbType() == POSTGRESQL ) {
+				assertThat( sqlTracker.getLoggedQueries() ).have( IS_PG_CREATE_TABLE_QUERY );
+			}
+		} );
+	}
+
+	@Test
+	public void testStringArrayTypeWithArrayAnotation(VertxTestContext context) {
 		Basic basic = new Basic();
 		String[] dataArray = {"Hello world!", "Hello earth"};
-		basic.stringArray = dataArray;
+		basic.stringArrayWithAnnotation = dataArray;
 
-		testField( context, basic, found -> assertArrayEquals( dataArray, found.stringArray ) );
+		testField( context, basic, found -> assertArrayEquals( dataArray, found.stringArrayWithAnnotation ) );
+	}
+
+	@Test
+	@DisabledFor(value = DB2, reason = "error executing SQL statement [An error occurred with a DB2 operation, SQLCODE=-302  SQLSTATE=22001]")
+	@DisabledFor(value = {MYSQL, MARIA}, reason = "errorMessage=Data too long for column 'stringArrayWithColumnAnnotation' at row 1, errorCode=1406, sqlState=22001")
```

Comment:
This error seems pretty explicit, the array is too big for this specific mapping. Can't you use a shorter array? Like `length = 5`?

Reply:
fixed and now all tests run except for ORACLE DB